cli/src/depsUpdate.ts

```diff
@@ -0,0 +1,128 @@
+import "zx/globals";
+import subprocess from "node:child_process";
+import {
+  convertToMarkdownTable,
+  createPR,
+  getLatestPackageVersion,
+} from "./util";
+
+export type DepsUpdateConfig = {
+  githubToken: string;
+  githubActor: string;
+};
+
+export async function depsUpdate({
+  githubToken,
+  githubActor,
+}: DepsUpdateConfig) {
+  const packages = getPackages();
+  const toUpdate = new Map<
+    string,
+    { packages: Record<string, string>; latestVersion: string }
+  >();
+  for (const pkg of packages) {
+    for (const dependencies of [
+      pkg.dependencies || {},
+      pkg.devDependencies || {},
+    ]) {
+      for (const [depName, info] of Object.entries(dependencies)) {
+        if (!info.resolved) {
+          // it's a local dependency, like the CLI -- not an NPM package
+          continue;
+        }
+        const existingUpdate = toUpdate.get(depName);
+        if (existingUpdate && existingUpdate.latestVersion !== info.version) {
+          Object.assign(existingUpdate.packages, {
+            [pkg.name]: info.version,
+          });
+          continue;
+        }
+        const latestVersion = await getLatestPackageVersion(depName);
+        if (latestVersion !== info.version) {
+          toUpdate.set(depName, {
+            packages: {
+              [pkg.name]: info.version,
+            },
+            latestVersion,
+          });
+        }
+      }
+    }
+  }
+
+  subprocess.execSync(`git config --global user.name "${githubActor}"`);
+  subprocess.execSync(
+    `git config --global user.email "${githubActor}@users.noreply.github.com"`,
+  );
+  const date = new Date().toISOString().slice(0, 10);
+  const depsToPRs = new Map();
+  for (const [depName, { packages, latestVersion }] of toUpdate) {
+    const head = `syncpack/${depName}-${date}`;
+    const base = "main";
+    const title = `syncpack update ${depName} to ${latestVersion}`;
+    echo(chalk.yellow(title));
+    const body = [
+      `Update ${depName} in the following packages:`,
+      ...Object.entries(packages).map(
+        ([packageName, version]) => `- ${packageName}: ${version}`,
+      ),
+    ].join("\n");
+    subprocess.execSync(`
+      git checkout -b ${head} ${base}
+      npx syncpack@alpha update --dependencies '${depName}'
+      pnpm install --no-frozen-lockfile --child-concurrency=10
+      pnpm run fix
+      if [[ -n "$(git diff --exit-code)" ]]; then
+        git add .
+        git commit -m '${title}'
+        git push
+      fi
+      `);
+    const { id, url } = await createPR({
+      githubToken,
+      head,
+      base,
+      title,
+      body,
+    });
+    depsToPRs.set(depName, `[#${id}](${url})`);
```

Comment:
So we'll have a PR per dependency? That will mean if we have to update multiple one week, we'll have to fix conflicts with main manually after merging each PR because the pnpm-lockfile will conflict, or no?

Reply:
Hm, yes. Though the alternative is bundling updates where only some work but some don't. Thoughts?